server/src/main/java/org/opensearch/action/support/clustermanager/ClusterManagerNodeReadRequest.java

```diff
@@ -54,12 +54,14 @@ protected ClusterManagerNodeReadRequest() {}
     protected ClusterManagerNodeReadRequest(StreamInput in) throws IOException {
         super(in);
         local = in.readBoolean();
+        shouldCancelOnTimeout = in.readBoolean();
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
         out.writeBoolean(local);
+        out.writeBoolean(shouldCancelOnTimeout);
```

Comment:
Ser/Deser has to be based on version-specific check. Otherwise it will break during upgrade.